src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -735,7 +735,13 @@ public static ActivityIdFormat DefaultIdFormat
             get
             {
                 if (s_defaultIdFormat == ActivityIdFormat.Unknown)
+                {
+#if NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT
```

Comment:
This define controls some functionality that seems unrelated what we are trying to do here. I'd recommend creating a new define in the .csproj file named something like "FEATURE_W3C_ON_BY_DEFAULT"

The current code enables W3C by default more broadly than I believe we want. My opinion is that we do not want the default to change for any .Net Framework app, which probably means limiting this change to netcoreapp TFM only. @davidfowl @shirhatti - not sure if you guys agree?

Reply:
This define is not per feature it is per targeting. I mean this is specific when targeting net45 and netstandard1.1. I think introducing a new definition may complicate what we already have. 

> The current code enables W3C by default more broadly than I believe we want. My opinion is that we do not want the default to change for any .Net Framework app, which probably means limiting this change to netcoreapp TFM only. @davidfowl @shirhatti - not sure if you guys agree?

I am fine to scope it to netcoreapp only if this is desired. I'll let @davidfowl @shirhatti advise about that.

Reply:
I'm inclined to agree with Noah here; I don't think we should make any changes to any Net Fx apps. I'd vouch for us to even more more conservative in our approaching to breaking an only target our changes for `net5.0`?

Can you shed some light on why we're filtering on this define `NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT`? Unless I'm mistaken, it seems unrelated.

Reply:
> I'm inclined to agree with Noah here; I don't think we should make any changes to any Net Fx apps. I'd vouch for us to even more more conservative in our approaching to breaking an only target our changes for net5.0?

I am going to change the implementation to restrict the change to .NET 5.0 only.

> Can you shed some light on why we're filtering on this define NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT? Unless I'm mistaken, it seems unrelated.

```XML
   <DefineConstants Condition="'$(TargetFramework)' == 'netstandard1.1' or '$(TargetFramework)' == 'net45'">$(DefineConstants);NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT</DefineConstants>
```
anyway, this point is now moot anyway as I am going to change the PR to restrict the change to .NET 5.0 only.